### PR TITLE
`prefetch_related_objects` fails to cache UUID FKs when the string representation of a UUID is used

### PR DESCRIPTION
**Pull request by selcuk**
# Trac ticket number

ticket-35434

# Branch description
Added unit test to reproduce the issue in the title.

# Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" ticket flag in the Trac system.
- [x] I have added or updated relevant tests.
- [ ] I have added or updated relevant docs, including release notes if applicable.
- [ ] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

File: tests/prefetch_related/test_uuid.py [hunk 1]
```diff
@@ -52,6 +53,14 @@ def test_prefetch_related_from_uuid_model_to_uuid_model_with_values_flat(self):
             [pet.id],
         )
 
+    def test_prefetch_related_objects_str_as_uuid_fk(self):
+        pet = Pet.objects.create(name="Fifi")
+        toy_ball = Toy(pet_id=pet.id, name="Ball")
+        toy_bone = Toy(pet_id=str(pet.id), name="Bone")
+        prefetch_related_objects([toy_ball, toy_bone], "pet")
+        self.assertEqual(pet, toy_ball.pet)
+        self.assertEqual(pet, toy_bone.pet)
```
Comment (jacobtylerwalls) [inline tests/prefetch_related/test_uuid.py:62]:
```suggestion
        with self.assertNumQueries(0):
            self.assertEqual(pet, toy_ball.pet)
            self.assertEqual(pet, toy_bone.pet)
```

**Review by jacobtylerwalls — COMMENTED** [review]
Thanks for the patch.

Had a look at whether the model with ManyToManyField could be used to test this and didn't find a way, so I think the new test model is fine.

One suggested change to clarify that the underlying functionality still works (and not merely doesn't crash).

Also verified this fixes the issue I encountered in my project (mentioned on ticket) 🚀 .

File: tests/prefetch_related/test_uuid.py (hunk 1 above)
Comment (selcuk) [reply tests/prefetch_related/test_uuid.py]: That makes perfect sense. Thanks, applied the suggestion.

**Comment by jacobtylerwalls** [comment]
I can mark this "ready for checkin" on Trac if you can address two things from the [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist): 

- commit message & pr title: use ticket number and period, like: Fixed #XXXXX -- Avoided bad thing.
- release note in docs/releases/5.0.5.txt. Although it's not a regression in a recent version (I had this problem on 4.2), `ObjectDoesNotExist` inappropriately escaping qualifies as a "crashing bug" under my understanding of the backport policy. Someone may correct me on this point.

**Pull request closed without merging**